### PR TITLE
Introduce 'getDistinctId' & 'identify' helpers

### PR DESCRIPTION
**Pull request by dimitrisnl**
Change-type: minor
Signed-off-by: Dimitrios Lytras <dimitrios@balena.io>

### CODE REVIEW

**Comment by lurch** [comment]
LOL, `identifyAnonymousUser` sounds like an oxymoron :wink:

**Comment by dimitrisnl** [comment]
You are right @lurch Let me update it :)

**Comment by gelbal** [comment]
Hey @dimitrisnl, does resin-api and/or balena-cli depend on this code? If so, how would that `isBrowser` check work for these repos? 

We have both CLI and API sending events to Mixpanel. They are not browsers so I just want to make sure they will keep on identifying the users fine. Honestly I know little about the implementation details here and thus my rather naive questions.

**Comment by dimitrisnl** [comment]
@gelbal 

These projects will not use any of these functions, so there should be no problem.
Since we're not editing existing functionality there should be no issue.

To give a little bit more context, mixpanel only exposes these functions for the browser sdk. https://www.npmjs.com/package/mixpanel#faq

**Comment by gelbal** [comment]
@dimitrisnl, got it. Thanks for the explanation.

File: src/resin-mixpanel-client.js [hunk 1]
```diff
@@ -110,6 +110,22 @@ module.exports = function(token, options) {
 
 				return mixpanel.track(event, props, callback)
 			})
+		},
+		getDistinctId: function() {
+			if (isBrowser) {
+				return mixpanel.get_distinct_id()
+			}
+			else {
```
Comment (thgreasi) [inline src/resin-mixpanel-client.js]: NIT: let's be consistent and use
```suggestion
			} else {
```
or even drop the `else`

File: src/resin-mixpanel-client.js [hunk 2]
```diff
@@ -110,6 +110,22 @@ module.exports = function(token, options) {
...
+				throw new Error('(Resin Mixpanel Client) function getDistinctId is only available for the browser')
+			}
+		},
+		identify: function(id) {
+			if (isBrowser) {
+				return mixpanel.identify(id)
+			}
+			else {
```
Comment (thgreasi) [inline src/resin-mixpanel-client.js]: Ditto

**Pull request merged**